name.abuchen.portfolio/src/name/abuchen/portfolio/util/WebAccess.java

```diff
@@ -5,78 +5,214 @@
         return this;
     }
 
     public WebAccess addHeader(String param, String value)
     {
+        param = Objects.requireNonNull(param, "PP WebAccess: Header: Parameter is null"); //$NON-NLS-1$
+        value = Objects.requireNonNull(value, "PP WebAccess: Header: Value Agent is null"); //$NON-NLS-1$
         this.headers.add(new BasicHeader(param, value));
         return this;
     }
 
+    public WebAccess addUserAgent(String userAgent)
+    {
+        this.userAgent = userAgent;
+        return this;
+    }
+
+    public WebAccess ignoreContentType(Boolean ignoreContentType)
+    {
+        this.ignoreContentType = ignoreContentType;
+        return this;
+    }
+
     public String get() throws IOException
     {
-        CloseableHttpClient client = HttpClientBuilder.create() //
-                        .setDefaultRequestConfig(defaultRequestConfig) //
-                        .setDefaultHeaders(this.headers) //
-                        .setUserAgent(OnlineHelper.getUserAgent()).build();
 
-        URIBuilder uriBuilder = new URIBuilder().setScheme(scheme).setHost(host).setPath(path);
-        uriBuilder.setParameters(this.parameters);
+        String errorURL = null;
+        CloseableHttpResponse response = null;
 
         try
         {
-            URL objectURL = uriBuilder.build().toURL();
-            CloseableHttpResponse response = client.execute(new HttpGet(objectURL.toString()));
-            if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK)
-                throw new IOException(objectURL.toString() + " --> " + response.getStatusLine().getStatusCode()); //$NON-NLS-1$
+            CloseableHttpClient client = HttpClientBuilder.create() //
+                            .setDefaultRequestConfig(defaultRequestConfig) //
+                            .setDefaultHeaders(this.headers) //
+                            .setUserAgent(this.userAgent).build();
+
+            if (this.url != null && !this.url.isEmpty())
+            {
+                response = client.execute(new HttpGet(url));
+                errorURL = this.url;
+            }
+            else
+            {
+                URIBuilder uriBuilder = new URIBuilder().setScheme(this.scheme).setHost(this.host).setPath(this.path);
+                uriBuilder.setParameters(this.parameters);
+                URL objectURL = uriBuilder.build().toURL();
+                response = client.execute(new HttpGet(objectURL.toString()));
+                errorURL = objectURL.toString();
+            }
+
+            if (!this.ignoreContentType)
+            {
+                HttpEntity entity = response.getEntity();
+                ContentType contentType;
+                if (entity != null)
+                {
+                    contentType = ContentType.get(entity);
+                    Boolean contentFound = filterContentType.stream()
+                                    .anyMatch(s -> contentType.getMimeType().toString().contains(s));
```

Comment:
Die Logik finde ich etwas zu komplex. Macht JSoup das so? Du könntest doch einfach schauen ob der MimeType in der Liste ist (List#contains(mimeType)) - warum schauen ob der MimeType in einem (Teil-)String eines der MimeTypes vorkommt?